app-modules/analytics/database/factories/AnalyticsResourceFactory.php

```diff
@@ -0,0 +1,75 @@
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Analytics\Database\Factories;
+
+use AdvisingApp\Analytics\Models\AnalyticsResource;
+use Illuminate\Database\Eloquent\Factories\Factory;
+use AdvisingApp\Analytics\Models\AnalyticsResourceSource;
+use AdvisingApp\Analytics\Models\AnalyticsResourceCategory;
+
+/**
+ * @extends Factory<AnalyticsResource>
+ */
+class AnalyticsResourceFactory extends Factory
+{
+    /**
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'name' => str(fake()->unique()->word())->ucfirst()->toString(),
+            'description' => fake()->sentences(asText: true),
+            'url' => fake()->optional()->url(),
+            'is_active' => fake()->boolean(),
+            'is_included_in_data_portal' => fake()->boolean(),
+        ];
+    }
+
+    public function configure(): AnalyticsResourceFactory|Factory
+    {
+        return $this->afterMaking(function (AnalyticsResource $analyticsResource) {
+            $analyticsResource
+                ->source()
+                ->associate(fake()->optional()->randomElement([AnalyticsResourceSource::inRandomOrder()->first() ?? AnalyticsResourceSource::factory()->create()]));
+
+            $analyticsResource
+                ->category()
+                ->associate(fake()->randomElement([AnalyticsResourceCategory::inRandomOrder()->first() ?? AnalyticsResourceCategory::factory()->create()]));
+        });
+    }
```

Comment:
Could we clean this up by just moving it to the `definition` method above? Like we do in `KnowledgeBaseItemFactory`?

Or is there are reason it needs to be in `configure`?